connect/src/com/telenor/connect/ConnectSdk.java

```diff
@@ -336,7 +336,13 @@ public static synchronized boolean isInitialized() {
 
     public static void logout() {
         Validator.sdkInitialized();
-        sdkProfile.getConnectIdService().revokeTokens(getContext());
+
+        if (sdkProfile instanceof MobileConnectSdkProfile) {
+            sdkProfile.getConnectIdService().revokeTokens(getContext());
+            return;
+        }
+
+        sdkProfile.getConnectIdService().logOut(getContext());
```

Comment:
Is it an option to add logout as part of SdkProfile interface.
Then you can just call sdkProfile.logout here,
for mc logout you could just: connectIdService.revokeTokens()
for connectid: connectIdService.logout

Reply:
Hmm... maybe... but I'm not sure that would make for cleaner code since the revoke and logOut functions needs the access token which, at least for connect id, is managed by `ConnectIdService` class (and this class is currently calling `connectApi.revokeToken(...)`.

Reply:
but that's fine , connectApi will point to a MobileConnectAPIAdapter in case of an MC profile and to ConnectAPI in case of a ConnectId. Or I'm not getting your concern correctly?